### PR TITLE
Don't force C++98 mode

### PR DESCRIPTION
**Pull request by gillins**
This is a bit of a test - latest GDAL is failing due to C++11 symbols not be found - I suspect it is because `kealib` forces C++98 mode. Disabling to see if it helps.

Also - using new format pinning to get same `hdf5` as GDAL - but this means disabling the GDAL test as there is no GDAL build with this version yet - chicken and egg situation!

Checklist
* [ ] Used a fork of the feedstock to propose changes
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy`
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by gillins** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -7,9 +7,12 @@ package:
 source:
     url: https://bitbucket.org/chchrsc/kealib/downloads/kealib-{{ version }}.tar.gz
     sha256: 1c80489f17114a229097c2e8c61d5e4c82ea63ae631c81a817fef95cfd527174
+    patches:
+      # by default kealib forces C++98 mode which breaks other things built in C++11 mode
+      - disablecpp98.patch
 
 build:
-    number: 0
+    number: 1
```
Comment (ocefpaf) [inline recipe/meta.yaml:15]: I guess it would be safer to remove the build number `0` to avoid conda picking it up.
I'll do that.

**Comment by gillins** [comment]
Good plan. I probably should have waited until you were online before tacking all this - sorry. Are we going to have the same hassle next time there is an `hdf5` upgrade?

**Comment by ocefpaf** [comment]
> Are we going to have the same hassle next time there is an `hdf5` upgrade?

Always :smile: but the move to `conda-build 3` should grant us less breakages.

**Pull request merged**